lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -186,11 +189,11 @@ def find_matches(self, requirements):
                 if all(self.is_satisfied_by(requirement, candidate) for requirement in requirements)
                 # FIXME
                 # if all(self.is_satisfied_by(requirement, candidate) and (
-                #     requirement.src is None or
+                #     requirement.src is None or  # if this is true for some candidates but not all it will break key param - Nonetype can't be compared to str
                 #     requirement.src == candidate.src
                 # ))
             },
-            key=operator.attrgetter('ver', 'src'),
+            key=lambda candidate: (SemanticVersion(candidate.ver), candidate.src,),
```

Comment:
This is the assertion in test_collection_install.py that caught this https://github.com/webknjaz/ansible/pull/354/files#diff-f98f4132c16dbdbc36a6b6481cf39d91a5a85e63210d72d535b2a5b63b013b32R616

Reply:
@s-hertel Do we know that `candidate.ver` is never `'*'`? Because that'd cause a `ValueError` as we know now.

Reply:
OTOH since it's coming from the Galaxy API, it's probably not going to happen.

Reply:
If it's coming from the Galaxy API then it should be semver, there's a test during import that validates the version is a proper semver string.